Src/Plugins/Security/PostgreSqlConnectionStringValidator.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Data.Common;
+using System.Text;
+
+using Microsoft.RE2.Managed;
+
+using Npgsql;
+
+namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security
+{
+    public class PostgreSqlConnectionStringValidator : ValidatorBase
+    {
+        internal static PostgreSqlConnectionStringValidator Instance;
+        internal static IRegex RegexEngine;
+
+        static PostgreSqlConnectionStringValidator()
+        {
+            Instance = new PostgreSqlConnectionStringValidator();
+            RegexEngine = RE2Regex.Instance;
+        }
+
+        public static string IsValidStatic(ref string matchedPattern,
+                                           ref Dictionary<string, string> groups,
+                                           ref string failureLevel,
+                                           ref string fingerprint,
+                                           ref string message)
+        {
+            return ValidatorBase.IsValidStatic(Instance,
+                                               ref matchedPattern,
+                                               ref groups,
+                                               ref failureLevel,
+                                               ref fingerprint,
+                                               ref message);
+        }
+
+        public static string IsValidDynamic(ref string fingerprint, ref string message)
+        {
+            return ValidatorBase.IsValidDynamic(Instance,
+                                                ref fingerprint,
+                                                ref message);
+        }
+
+        protected override string IsValidStaticHelper(ref string matchedPattern, ref Dictionary<string, string> groups, ref string failureLevel, ref string fingerprintText, ref string message)
```

Comment:
>ref string matchedPattern, ref Dictionary<string, string> groups, ref string failureLevel, ref string fingerprintText, ref string message [](start = 54, length = 137)

wrap the parameters, pls

Reply:
Done

---
In reply to: [568863692](https://github.com/microsoft/sarif-pattern-matcher/pull/157#discussion_r568863692) [](ancestors = 568863692)